internal/beater/server_test.go

```diff
@@ -630,15 +628,12 @@ func TestWrapServerAPMInstrumentationTimeout(t *testing.T) {
 		}
 	}
 	// Assert that metrics have expected response values reported.
-	equal, result := monitoringtest.CompareMonitoringInt(map[request.ResultID]int{
-		request.IDRequestCount:          2,
-		request.IDResponseCount:         2,
-		request.IDResponseErrorsCount:   1,
-		request.IDResponseValidCount:    1,
-		request.IDResponseErrorsTimeout: 1, // test data POST /intake/v2/events
-		request.IDResponseValidAccepted: 1, // self-instrumentation
-	}, intake.MonitoringMap)
-	assert.True(t, equal, result)
+	monitoringtest.ExpectContainOtelMetrics(t, reader, map[string]any{
+		"http.server." + string(request.IDRequestCount):          1,
+		"http.server." + string(request.IDResponseCount):         1,
+		"http.server." + string(request.IDResponseErrorsCount):   1,
+		"http.server." + string(request.IDResponseErrorsTimeout): 1, // test data POST /intake/v2/events
+	})
```

Comment:
Probably worth adding a comment that the self-instrumentation requests are not counted in metrics.

This is a change in behaviour. I think it's probably what users would want, but would like to hear @simitt's opinion on this.

Reply:
I agree that we don't want to mingle the counters for user requests and self-instrumentation requests together, especially when not being able to encode a differentiator in the metadata. Seems like a bugfix to me.